src/distiller/distiller.py

```
@@ -248,35 +251,64 @@ def _get_concepts(self, doc_id: str) -> list[dict]:
+
+        Phase 2: Fetch full relationship data only for top_k concepts.
+        Caches in ``_related`` so ``_build_prerequisites`` can reuse it.
         """
+        # Phase 1: Count edges concurrently for all concepts
+        def _count_edges(concept: dict) -> tuple[dict, int]:
+            internal_id = concept.get("id")
+            if not internal_id:
+                return (concept, 0)
+            try:
+                count = self.store.count_concept_edges_direct(internal_id)
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)